lib/jsdom/browser/Window.js

```diff
@@ -154,6 +154,15 @@ function Window(options) {
       lastUrl = fixUrlForBuggyXhr(resolveHref(window.document.URL, url));
       return xhr._open(method, lastUrl, async, user, password);
     };
+    xhr._getAllResponseHeaders = xhr.getAllResponseHeaders;
+    xhr.getAllResponseHeaders = function () {
+      if (window.document.location.protocol === "file:") {
```

Comment:
I assume the breakage actually happens when lastUrl is file, not when the window location is file.